### PR TITLE
web3 and function calls

### PR DESCRIPTION
**Pull request by chaitanya5**
The objective is to create an interface to sell the Loser Boxes.

This interface will contain these features:

- Sale via an escrow smart contract (Kleros)
- The currency of exchange is the DAI (50 DAI per Box)
- There will be a swapper to change its currency if necessary (dex.ag).

The documentation of the stepper is here: https://material-ui.com/components/steppers/.

**The interface will contain a stepper composed of 5 steps:**

1. Fill your personal details
2. Connect to Web3 (optional)
3. Swap to DAI (optional) => use dex.ag SDK, see https://docs.dex.ag/sdk
4. Approve to DAI
5. Transfer DAI to the escrow

After the last transaction redirects to the confirmation page.

**The User Interface is described here:** 

https://www.figma.com/file/a4C0WETqHQR4siSgnRNnn1b7/RECOVER?node-id=0%3A1.

The color of the dots stepper must be #12c2e9 for an inactive step and #a6ffcb for the active step.

For the transaction message box, here is the code source: https://github.com/blockchain-mafia/recoverto-UI/blob/master/src/components/message-box-tx.js.

The confirmation/approval button will be present on all pages except the skipped one (that will depend on the connection to web3).

The button “Buy a Loser Box” redirects to the form page.
The form page has to be a form with a pre-validation form.

We have to push the contract generated with the personal details of the client to IPFS (ipfs.kleros.io).

Make this page responsive.

**Deadline**
31 December 2020.
Total time: 50h.

### Before at  ./pages/loser-box.js
**Line 136** 
` const erc20ContractAddress = '0x75A27fa3F7ec1ECB1fd05618DB5Fab78de0755D2';`
**Line 141**
` const tokenAmount = 10;`

### After at ./pages/loser-box.js
**Line 136** 
` const erc20ContractAddress = 'YOUR ERC20 TOKEN ADDRESS';`
**Line 141**
` const tokenAmount = YOUR REQUIRED AMOUNT;`

### CODE REVIEW

**Comment by n1c01a5** [comment]
@chaitanya5 can you respect the "snake case" format to name the files.

**Comment by n1c01a5** [comment]
@chaitanya5 there is an issue with the alignement of the "Recipient's name" field.

**Comment by n1c01a5** [comment]
@chaitanya5 the second step has to be "optional" as well.

**Comment by n1c01a5** [comment]
As discussed, the `connect Web3` will be not optional.

**Comment by n1c01a5** [comment]
@chaitanya5 there is too much space between the title and the content.

**Comment by n1c01a5** [comment]
@chaitanya5 don't put `*` and `(Optional)`. `(optional)` in lower case is enough.

**Comment by n1c01a5** [comment]
@chaitanya5 can put th elabel under the dot like https://codesandbox.io/s/3v8gs .

**Comment by n1c01a5** [comment]
@chaitanya5 also the header is not the same than the other page. You have to re-use the header to be consistant.

**Comment by chaitanya5** [comment]
@n1c01a5 , thank you for updating here. I was following whatever was shown in the Figma. I will make sure to implement it. 
Also please check the form validations thoroughly and let me know the changes. It's been a while changing the form and error validations from time to time. 
For convinience, let me summarise everthing,

1. can you respect the "kebab case" format to name the files.
2. there is too much space between the title and the content.
3. don't put * and (Optional). (optional) in lower case is enough.
4. can put th elabel under the dot like https://codesandbox.io/s/3v8gs .
5. also the header is not the same than the other page. You have to re-use the header to be consistant.

**Comment by bhavesh92pandey** [comment]
@chaitanya5 Yes, there was (*) present in Figma design which I have removed now.

**Comment by bhavesh92pandey** [comment]
In progress #65

**Review by n1c01a5 — CHANGES_REQUESTED** [review]

**Comment by n1c01a5** [comment]
@chaitanya5 can you fix the conflicts

File: pages/loserbox-stepper.js [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+          .call()
+        if (userTokenBalance > value) {
+          setTokenBalanceApproved(true)
+        }
+        handleNext(networkId)
+      }
+    }
+  }
+
+  const handleNext = (networkId) => {
```
Comment (n1c01a5) [inline pages/loserbox-stepper.js:206]: It's more something like `handleNextStep` to make explicit it's connected with the stepper.

File: pages/loserbox-stepper.js [hunk 2]
```diff
@@ -0,0 +1,342 @@
...
+    if (networkId === 42 || networkName !== '') {
+      queryName = { network: 'kovan', step: encodeURI(pageCount) }
+    } else {
+      queryName = { step: encodeURI(pageCount) }
+    }
+    router.push({
+      pathname: '/loserbox-stepper',
+      query: queryName,
+    })
+    console.log('activeStepNext', activeStep + 1)
```
Comment (n1c01a5) [inline pages/loserbox-stepper.js:219]: Remove this log.

File: pages/loserbox-stepper.js [hunk 3]
```diff
@@ -0,0 +1,342 @@
...
+function getSteps() {
+  return [
+    'Personal Details',
+    'Connect to Web3',
+    `Swap ETH to ${process.env.NEXT_PUBLIC_MAINNET_TOKEN_AMOUNT} DAI`,
+    'Approve DAI Transfer',
+    'Transfer DAI to the Escrow',
+  ]
+}
+let pageCount
```
Comment (n1c01a5) [inline pages/loserbox-stepper.js:41]: It looks weird to have this.
Is it not possible to solve the state stepper with the `activeStap useState`?

File: pages/pay-loserbox.js [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+          )
+          amount = web3.utils.toBN(process.env.NEXT_PUBLIC_KOVAN_TOKEN_AMOUNT)
+        }
+        let value = amount.mul(web3.utils.toBN(10).pow(decimals))
+
+        const TxId = await mattContract.methods
+          .getTransactionIDsByAddress(accounts[0])
+          .call();
+
+        console.log('TxId', TxId)
```
Comment (n1c01a5) [inline pages/pay-loserbox.js:95]: Cna you remove all logs like this.

**Review by n1c01a5 — CHANGES_REQUESTED** [review]

**Comment by n1c01a5** [comment]
@chaitanya5 fix the conflicts!

File: components/stepper/approve-dai.js [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+    setIsPending(true)
+    setIsOngoing(false)
+    setButtonView(false)
+
+    const data = tokenContract.methods
+      .approve(envData.MULTIPLE_ARBITRABLE_CONTRACT_ADDRESS, tokenAmount)
+      .encodeABI()
+
+    const transactionParameters = {
+      to: envData.ERCTOKEN, // Required except during contract publications.
```
Comment (bhavesh92pandey) [inline components/stepper/approve-dai.js:50]: ERC_TOKEN

File: components/stepper/approve-dai.js [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+      .once('confirmation', (confirmationNumber, receipt) => {
+        if (receipt.status) {
+          submitPersonalDetails()
+        }
+      })
+      .on('error', (error) => {
+        console.error('error', error)
+        setIsOngoing(false)
+        setIsPending(false)
+        setTxerror(true)
```
Comment (bhavesh92pandey) [inline components/stepper/approve-dai.js:70]: setTXError

File: components/stepper/connect-web3.js [hunk 7]
```diff
@@ -0,0 +1,68 @@
+import React from 'react'
+
+export default function ConnectWeb3 ({
+  isnetworkWarning,
```
Comment (bhavesh92pandey) [inline components/stepper/connect-web3.js:4]: isNetworkWarning

File: components/stepper/connect-web3.js [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+  return (
+    <div style={{ paddingTop: 50 }}>
+      <div className='row form-group' style={{ padding: '.375rem .75rem' }}>
+        <h4>
+          <span style={{ color: '#13a2dc' }}>Connect</span> to Web3
+        </h4>
+      </div>
+      <div className='row'>
+        <div className='col-md-12'>
+          {isnetworkWarning
```
Comment (bhavesh92pandey) [inline components/stepper/connect-web3.js:17]: isNetworkWarning

File: components/stepper/transfer-dai.js [hunk 9]
```diff
@@ -0,0 +1,238 @@
...
+  const handleClickOpen = (event) => {
+    event.preventDefault()
+    setOpen(true)
+  }
+
+  const handleClose = () => {
+    setOpen(false)
+  }
+  const transfer = async () => {
+    if (isagree) {
```
Comment (bhavesh92pandey) [inline components/stepper/transfer-dai.js:43]: isAgree

File: components/stepper/transfer-dai.js [hunk 10]
```diff
@@ -0,0 +1,238 @@
...
+                  <div>
+                    {isOngoing ? <BounceLoader size={50} color='#fff' /> : null}
+                  </div>
+                </div>
+              </div>
+              )
+            : null}
+        </div>
+      </div>
+      {open && (
```
Comment (bhavesh92pandey) [inline components/stepper/transfer-dai.js:167]: isOpen

**Review by bhavesh92pandey — CHANGES_REQUESTED** [review]

**Comment by n1c01a5** [comment]
@chaitanya5 yes good think to use the `dex sdk`.

**Comment by bhavesh92pandey** [comment]
Please resolve these conflicts as discussed

**Pull request merged**